networkx/algorithms/maximum_weight_cycle_packing.py

```diff
@@ -0,0 +1,204 @@
+        result.append(cycle)
+
+    return result  # exchanges
+
+
+def connect_2cycles(X, graph, ans_graph):
+    for i in range(
+        len(X)
+    ):  # creating the edges in the graph by going through the 2-cycles
+        for j in range(i + 1, len(X)):
+            if (X[i], X[j]) in graph.edges and (X[j], X[i]) in graph.edges:
+                weight = (
+                    graph.get_edge_data(X[i], X[j])["weight"]
+                    + graph.get_edge_data(X[j], X[i])["weight"]
+                )
+                ans_graph.add_edge((X[i]), (X[j]), weight=weight, cycle=[X[i], X[j]])
+
+
+def connect_3cycles(X, Y, graph, ans_graph):
+    #   creating the edges in the graph by going through the 3-cycles
+    for k in range(len(X)):
+        for j, l in Y:  # This deals with the normal case of Yi,j Xk
+            if (l, X[k]) in graph.edges and (X[k], j) in graph.edges:
+                weight = (
+                    graph.get_edge_data(j, l)["weight"]
+                    + graph.get_edge_data(l, X[k])["weight"]
+                    + graph.get_edge_data(X[k], j)["weight"]
+                )
+                ans_graph.add_edge((X[k]), (j, l), weight=weight, cycle=[j, l, X[k]])
+
+
+def check_independent(temp):
+    """This function is used to check if a list of edges are independent set
+    >>> check_independent([(2,3),(1,4),(5,7)])
+    True
+    >>> check_independent([(2,3),(1,4),(5,1)])
+    False
+    """
+    nodes_seen = []
+    for cyc in temp:
+        for node in cyc:
+            if node in nodes_seen:
+                return False
+            nodes_seen.append(node)
+    return True
+
+
+def create_Ys(graph, k):
+    """This function is used to create the cartesian product of the 3-cycles
+    >>> Digraph=nx.DiGraph()
+    >>> Digraph.add_nodes_from([1,2,3,5,6,7,8])
+    >>> Digraph.add_weighted_edges_from([(1,8,2),(8,1,4),(2,1,5),(1,3,4),(3,8,2),(8,2,3),(8,5,4),(5,7,3),(7,6,2),(6,5,4)])
+    >>> Ys,_=create_Ys(Digraph,3)
+    >>> print(((len(Ys)==6 ) or (len(Ys) == 8))) #- the known product is supposed to be composed of 27 permutation
+    True
+    >>> Digraph =nx.DiGraph()
+    >>> Digraph.add_nodes_from([1,2,3,4])
+    >>> Digraph.add_weighted_edges_from([(2,1,3),(1,3,1),(3,2,2),(3,4,5),(4,3,9)])
+    >>> print(len(create_Ys(Digraph,3))) #- the known product is supposed to be composed of 1 permutation
+    2
+    """
+
+    import itertools
+    import random
+
+    import numpy as np
+
+    temp_cycles = simple_cycles_le_k(graph, k)
+    cycles = []
+    for cycle in temp_cycles:
+        if len(cycle) == k:
+            cycles.append(cycle)
+
+    perm_arr = np.ndarray(shape=(len(cycles), k), dtype=list)
+    for cyc_idx in range(len(cycles)):
+        cyc = cycles[cyc_idx]
+        for ed_idx in range(len(cyc)):
+            mid = (cyc[ed_idx], cyc[(ed_idx + 1) % len(cyc)])
+            perm_arr[cyc_idx][ed_idx] = mid
+    S = []
+    for cyc in perm_arr:
+        rand_num = random.randint(0, len(cyc) - 1)
```

Comment:
This does not have to be random - it can be arbitrary. You can always take e.g. cyc[0].

Reply:
Also, you can initialize S=set(), and just append cyc[0] (if it already exists, it will not be added again).